packages/admin-panel/src/pages/resources/CustomLandingPagesPage.js

```diff
@@ -285,6 +285,13 @@ export const CustomLandingPagesPage = ({ getHeaderEl }) => {
       columns={COLUMNS}
       getHeaderEl={getHeaderEl}
       createConfig={CREATE_CONFIG}
+      onProcessDataForSave={(editedFields, recordData) => {
+        // If the landing page is being edited, and the url_segment field is not being edited, then include the existing url_segment in the edited fields so that it can be used for generating landing page image names.
```

Comment:
I see the intent but the server should be able to pull this up from the existing record, I'd say doing it server side will be clearer than in the frontend. Also api-first thinking means that in the future if another client wants to call this same API the behaviour will not need to be reimplemented.